src/WebJobs.Script.WebHost/Configuration/HttpBodyControlOptionsSetup.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Configuration
+{
+    internal class HttpBodyControlOptionsSetup : IConfigureOptions<HttpBodyControlOptions>
+    {
+        private readonly IEnvironment _environment;
+
+        public HttpBodyControlOptionsSetup(IEnvironment environment)
+        {
+            _environment = environment;
+        }
+
+        public void Configure(HttpBodyControlOptions options)
+        {
+            options.AllowSynchronousIO = _environment.IsV2CompatibilityMode()
+                || FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagAllowSynchronousIO, _environment);
+        }
```

Comment:
I decided to remove the `CompatibilityModeOptions` and it's setup in favor of having more specific options -- and having those options check whether they're in compat mode. Having two Options objects that could depend on each other and both need to reset via a change token had some issues with timing.

Reply:
👍